kedro/extras/datasets/pandas/sql_dataset.py

```diff
@@ -374,32 +385,48 @@ def __init__(  # pylint: disable=too-many-arguments
             self._protocol = protocol
             self._fs = fsspec.filesystem(self._protocol, **_fs_credentials, **_fs_args)
             self._filepath = path
-        self._load_args["con"] = credentials["con"]
+        self._connection_str = credentials["con"]
+        self.create_connection(self._connection_str)
+
+    @classmethod
+    def create_connection(cls, connection_str: str) -> None:
+        """Given a connection string, create singleton connection
+        to be used across all instances of `SQLQueryDataSet` that
+        need to connect to the same source.
+        """
+        if connection_str in getattr(cls, "engines", {}):
+            return  # pragma: no cover
```

Comment:
I added `test_create_connection_only_once` unit test specifically to cover this but it still doesn't get captured in coverage, so added a `pragma: no cover` too.

Reply:
Isn't that because the test uses the `SQLTableDataSet` and not the `SQLQueryDataSet` ?

Reply:
🤦‍♀️  I didn't pay close attention, probably... I'll check, thanks!